com.microsoft.mrtk.uxcore/Keyboard/NonNative/NonNativeKeyboard.cs

```diff
@@ -0,0 +1,732 @@
+            InputField.text = InputField.text.Insert(m_CaretPosition, tabString);
+            m_CaretPosition += tabString.Length;
+
+            UpdateCaretPosition(m_CaretPosition);
+        }
+
+        /// <summary>
+        /// Insert a tab character.
+        /// </summary>
+        public void MoveCaretRight()
+        {
+            m_CaretPosition = InputField.caretPosition;
+
+            if (m_CaretPosition < InputField.text.Length)
+            {
+                ++m_CaretPosition;
+                UpdateCaretPosition(m_CaretPosition);
+            }
+        }
+
+        /// <summary>
+        /// Insert a tab character.
+        /// </summary>
+        public void MoveCaretLeft()
+        {
+            m_CaretPosition = InputField.caretPosition;
+
+            if (m_CaretPosition > 0)
+            {
+                --m_CaretPosition;
+                UpdateCaretPosition(m_CaretPosition);
+            }
+        }
+
+        /// <summary>
+        /// Close the keyboard.
+        /// </summary>
+        public void Close()
+        {
+            if (InputField != null)
+            {
+                OnClose.Invoke(InputField.text);
+                gameObject.SetActive(false);
+            }
+        }
+
+        /// <summary>
+        /// Clear the text field and reset keyboard state (e.g. Shift and CapsLock).
+        /// </summary>
+        public void Clear()
+        {
+            if (InputField != null)
+            {
+                ResetKeyboardState();
+                InputField.text = "";
+                m_CaretPosition = InputField.caretPosition;
+            }
+        }
+
+        #endregion Keyboard Functions
+
+        #region Keyboard Layout Modes
+
+        private void ShowAlphaKeyboardUpperSection()
+        {
+            alphaKeysSection?.SetActive(true);
+        }
+
+        private void ShowAlphaKeyboardDefaultBottomKeysSection()
+        {
+            if (defaultBottomKeysSection?.transform.parent.gameObject.activeSelf == false)
+            {
+                defaultBottomKeysSection?.transform.parent.gameObject.SetActive(true);
+            }
+            defaultBottomKeysSection?.SetActive(true);
+        }
+
+        private void ShowAlphaKeyboardEmailBottomKeysSection()
+        {
+            if (emailBottomKeysSection?.transform.parent.gameObject.activeSelf == false)
+            {
+                emailBottomKeysSection?.transform.parent.gameObject.SetActive(true);
+            }
+            emailBottomKeysSection?.SetActive(true);
+        }
+
+        private void ShowAlphaKeyboardURLBottomKeysSection()
+        {
+            if (urlBottomKeysSection?.transform.parent.gameObject.activeSelf == false)
+            {
+                urlBottomKeysSection?.transform.parent.gameObject.SetActive(true);
+            }
+            urlBottomKeysSection?.SetActive(true);
+        }
+
+        private void ShowSymbolKeyboard()
+        {
+            symbolKeysSection?.gameObject.SetActive(true);
```

Comment:
change "?" to "symbolKeysSection != null"  since the "?" operator doesn't fully work with MonoBehaviours and GameObjects....same comment for all other usages of "?"

Reply:
will do!